lib/nostrum/gateway/hello_event.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Nostrum.Gateway.HelloEvent do
+  defstruct [
+    :heartbeat_interval,
+    :_trace
+  ]
+
+  defmodule Field do
+    @moduledoc false
+
+    @spec new({term, term}) :: Keyword.t()
+    def new(pair)
+    def new({k, v}) when not is_binary(k), do: new({to_string(k), v})
+    def new({"heartbeat_interval", value}), do: [heartbeat_interval: value]
```

Comment:
What's the reasoning here behind converting these to a kwl? Couldn't the same be done with a map?